src/DependencyInjection/Compiler/TypeRegistryPass.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MongoDBBundle\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\MongoDBBundle\Attribute\AsFieldType;
+use Doctrine\Bundle\MongoDBBundle\Types\LazyTypeRegistry;
+use InvalidArgumentException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\Compiler\ServiceLocatorTagPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+use function array_replace;
+use function array_reverse;
+use function sprintf;
+
+/**
+ * Register all types as services in the type registry.
+ *
+ * @internal
+ */
+final class TypeRegistryPass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    private const TAG = 'doctrine_mongodb.odm.field_type';
+
+    private const ALL = '\0';
```

Comment:
I'll note that this is not actually a null byte, but rather the literal string `\0`. I would hope people don't actually use that.

Reply:
Null byte is better, isn't it?
```suggestion
    private const ALL = "\0";
```

But this can be anything that is not a valid document manager name.